sequencer/src/l1_client.rs

```diff
@@ -276,140 +163,70 @@ async fn get_finalized_block<P: JsonRpcClient>(
+        let l1_client = L1Client::new(anvil.url());
+        let provider = l1_client.provider;
 
-            let expected_head = anvil_client.get_block_number().await.unwrap().as_u64();
-            let expected_finalized = anvil_client
-                .get_block(BlockNumber::Finalized)
-                .await
-                .unwrap();
+        let expected_head = provider.get_block_number().await.unwrap().as_u64();
+        let expected_finalized = provider.get_block(BlockNumber::Finalized).await.unwrap();
 
-            // The snapshotted head should be within one of the observed head, it may be one block
-            // behind if the head just changed and the snapshot hasn't updated yet.
-            assert!(
-                expected_head == snapshot.head || expected_head == snapshot.head + 1,
-                "snapshotted head {} does not match expected head {}",
-                snapshot.head,
-                expected_head
-            );
-
-            // And same for the finalized block, unless there is no finalized block.
-            if let Some(expected_finalized) = expected_finalized {
-                let expected_finalized_number = expected_finalized.number.unwrap().as_u64();
-                let snapshot_finalized = snapshot.finalized.unwrap();
-                assert!(
-                    expected_finalized_number == snapshot_finalized.number
-                        || expected_finalized_number == snapshot_finalized.number + 1,
-                    "snapshotted finalized {} does not match expected finalized {}",
-                    snapshot_finalized.number,
-                    expected_finalized_number
-                );
-                if expected_finalized_number == snapshot_finalized.number {
-                    assert_eq!(expected_finalized.hash.unwrap(), snapshot_finalized.hash);
-                    assert_eq!(expected_finalized.timestamp, snapshot_finalized.timestamp);
-                    tracing::info!("L1 finalized block fully synchronized");
-                }
-            } else {
-                tracing::info!("no finalized L1 block yet");
-                assert_eq!(snapshot.finalized, None);
-            }
+        assert_eq!(7, expected_head);
 
-            // Check invariants: monitonically increasing L1 blocks.
-            assert!(snapshot.head >= prev_head);
-            if let Some(finalized) = snapshot.finalized {
-                if let Some(prev_finalized) = prev_finalized {
-                    assert!(finalized.number >= prev_finalized.number);
-                }
-            } else {
-                assert_eq!(prev_finalized, None);
-            }
+        let finalized = get_finalized_block(&provider).await.unwrap().unwrap();
 
-            prev_head = snapshot.head;
-            prev_finalized = snapshot.finalized;
+        // Though we are asserting a values, we do so to prove
+        // structural correctness.
+        assert_eq!(0, finalized.number);
```

Comment:
Don't understand the comment. Also not sure about hard-coding 0 here. Isn't it based on timing?

Reply:
In ef28a828d766970c14c8d2b42b1b2afbadd4347e I've removed all the timing based assertions which also removes this comment.